swap/src/bob/swap.rs

```diff
@@ -359,6 +364,31 @@ where
                 )
                 .await
             }
+            BobState::T1Expired(state4) => {
+                if state4
+                    .check_for_tx_cancel(bitcoin_wallet.as_ref())
+                    .await
+                    .is_err()
+                {
+                    state4.submit_tx_cancel(bitcoin_wallet.as_ref()).await?;
```

Comment:
Once #97 is done, then we should just try to broadcast, no need to verify beforehand.

Reply:
That is true #97 actually branched off this branch so the functionality is added on both sides :)